src/Provider/UploadServiceProvider.php

```diff
@@ -18,6 +19,13 @@ class UploadServiceProvider implements ServiceProviderInterface
 {
     public function register(Application $app)
     {
+        $app['upload.sanitizer'] = $app->share(function ($app) {
+            $pattern = $app['config']->get('general/upload/pattern', '[^A-Za-z0-9\.]+');
+            $lowercase = $app['config']->get('general/upload/lowercase', true);
+
+            return new Slugify("/$pattern/", ['lowercase' => $lowercase]);
```

Comment:
@GawainLynch i wasn't sure, how much memory does Slugify take and is it only 1 instance per PHP response called. I mean, if multiple files and one php session, and heavy Slugify. So that's why i created distinct variables. This solution was my first one. And thanks for using that.

Reply:
Yeah, Slugify is tiny … there are much bigger maps in use. But also this is the DI service model, they only get instantiated once per request as you mention :smile: